code/tests/test_app.py

```diff
@@ -283,12 +302,120 @@ def test_converstation_azure_byod_returns_correct_response_when_streaming_with_d
         data = str(response.data, "utf-8")
         assert (
             data
-            == """{"id": "response.id", "model": "some-model", "created": 0, "object": "response.object", "choices": [{"messages": [{"content": "{\\"citations\\": [{\\"content\\": \\"content\\", \\"title\\": \\"title\\"}], \\"intent\\": \\"intent\\"}", "end_turn": false, "role": "tool"}, {"content": "", "end_turn": false, "role": "assistant"}]}]}
-{"id": "response.id", "model": "some-model", "created": 0, "object": "response.object", "choices": [{"messages": [{"content": "{\\"citations\\": [{\\"content\\": \\"content\\", \\"title\\": \\"title\\"}], \\"intent\\": \\"intent\\"}", "end_turn": false, "role": "tool"}, {"content": "A question\\n?", "end_turn": false, "role": "assistant"}]}]}
-{"id": "response.id", "model": "some-model", "created": 0, "object": "response.object", "choices": [{"messages": [{"content": "{\\"citations\\": [{\\"content\\": \\"content\\", \\"title\\": \\"title\\"}], \\"intent\\": \\"intent\\"}", "end_turn": false, "role": "tool"}, {"content": "A question\\n?", "end_turn": true, "role": "assistant"}]}]}
+            == r"""{"id": "response.id", "model": "some-model", "created": 0, "object": "response.object", "choices": [{"messages": [{"content": "{\"citations\": [{\"content\": \"content\", \"title\": \"title\"}], \"intent\": \"intent\"}", "end_turn": false, "role": "tool"}, {"content": "", "end_turn": false, "role": "assistant"}]}]}
+{"id": "response.id", "model": "some-model", "created": 0, "object": "response.object", "choices": [{"messages": [{"content": "{\"citations\": [{\"content\": \"content\", \"title\": \"title\"}], \"intent\": \"intent\"}", "end_turn": false, "role": "tool"}, {"content": "A question\n?", "end_turn": false, "role": "assistant"}]}]}
+{"id": "response.id", "model": "some-model", "created": 0, "object": "response.object", "choices": [{"messages": [{"content": "{\"citations\": [{\"content\": \"content\", \"title\": \"title\"}], \"intent\": \"intent\"}", "end_turn": false, "role": "tool"}, {"content": "A question\n?", "end_turn": true, "role": "assistant"}]}]}
 """
         )
 
+        request_body = mock_session.post.call_args[1]["json"]
+        request_headers = mock_session.post.call_args[1]["headers"]
+
+        assert request_body["data_sources"][0]["parameters"]["authentication"] == {
+            "type": "api_key",
+            "key": "some-search-key",
+        }
+        assert request_headers["api-key"] == "some-api-key"
```

Comment:
Could you use the already defined variables in the assertions? E.g.
```suggestion
        assert request_headers["api-key"] == self.openai_api_key
```
Should make the tests easier to maintain in the future.

Reply:
I've updated the majority of them, hopefully that's fine